tubular/comparison.py

```diff
@@ -72,6 +72,7 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         X[self.new_col_name] = X[self.columns[0]] == X[self.columns[1]]
 
         if self.drop_original:
-            X = X.drop(self.columns, axis=1)
+            for col in self.columns:
```

Comment:
I might be wrong but, wouldn't shifting this logic for dropping columns to BaseTransformer.transform then delete these columns during the X = super().transform(X) call? In that case, in the code line: X[self.new_col_name] = X[self.columns[0]] == X[self.columns[1]],  the columns in self.columns won't be accessible since they would have already been deleted from X. Let me know what you think.